File: L861-ScoreAfterFlippingMatrix.java
// https://leetcode.com/problems/score-after-flipping-matrix/
class Solution {
    public int matrixScore(int[][] grid) {
        int ROW = grid.length, COL = grid[0].length, half = ROW / 2;
        int res = (1 << (COL - 1)) * ROW;
        for (int row = 0; row < ROW; row++) {
            if (grid[row][0] == 0) {
                for (int col = 0; col < COL; col++) {
                    grid[row][col] ^= 1;
                }
            }
        }
        for (int col = 1; col < COL; col++) {
            int zeroCount = 0, curScore = 0;
            for (int row = 0; row < ROW; row++) {
                if (grid[row][col] == 0) {
                    zeroCount += 1;
                } else {
                    curScore += (1 << (COL - col - 1));
                }
            }
            if (zeroCount > half) {
                for (int row = 0; row < ROW; row++) {
                    grid[row][col] ^= 1;
                    res += (grid[row][col] << (COL - col - 1));
                }
            } else {
                res += curScore;
            }
        }
        return res;
    }
}

File: L861-score_after_flipping_matrix.py
# https://leetcode.com/problems/score-after-flipping-matrix/
class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        m = len(grid) # row
        n = len(grid[0]) # column
        res = (1 << (n - 1)) * m
        for row in range(m):
            if (grid[row][0] == 0):
                for col in range(n):
                    grid[row][col] ^= 1
        for col in range(1, n):
            zeroCount, curScore = 0, 0
            for row in range(m):
                if (grid[row][col] == 0):
                    zeroCount += 1
                else:
                    curScore += (1 << (n - col - 1))
            if (zeroCount > m // 2):
                for row in range(m):
                    grid[row][col] ^= 1
                    res += (grid[row][col] << (n - col - 1))
            else:
                res += curScore
        return res
